src/client/cli/cmd/launch.cpp

```diff
@@ -277,12 +279,36 @@ mp::ParseCode cmd::Launch::parse_args(mp::ArgParser* parser)
 
     if (parser->isSet(memOption))
     {
-        request.set_mem_size(parser->value(memOption).toStdString());
+        auto arg_mem_size = parser->value(memOption).toStdString();
+
+        try
+        {
+            mp::MemorySize{arg_mem_size};
+        }
+        catch (mp::InvalidMemorySizeException&)
+        {
+            fmt::print(cerr, "Invalid memory size value supplied: {}.\n", arg_mem_size);
+            return ParseCode::CommandLineError;
+        }
```

Comment:
Hmm, the exception already includes ~same info, and it is handled "above" if you just throw. So, this would be enough:

```suggestion
        mp::MemorySize{arg_mem_size}; // throw if bad
```

See https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#e17-dont-try-to-catch-every-exception-in-every-function and https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#e18-minimize-the-use-of-explicit-trycatch .

This yields:

```
[2021-09-10T11:53:57.777] [error] [client] Caught an unhandled exception: asdf is not a valid memory size
```

As opposed to:

```
Invalid disk size value supplied: asdf.
```

I think the simplicity is preferable here, but if you prefer the tailored message, I am also OK with that. Up to you.